source/router.ts

```diff
@@ -249,6 +249,11 @@ export class Router extends EventTarget {
 		for (let path in routes) {
 			const route = routes[path];
 
+			// the default route in the root layer should be '/' instead of '' as an empty string isn't a valid full path
```

Comment:
isn't this solving a symptom? fix seems at an odd location

Reply:
I have tried to change the getActivePath for the path router but this triggered a lot of other issues in page-default-directives and more...

basically when the router navigates to the '' location it can't history.pushState as it doesn't respect an empty string. it must be '/'. I thought that this approach would be cleaner and simpler.

Reply:
does the path router get a '' or a '/'? this is a path-router specific problem, the hash router or other router implementations do not require a '/' at the beginning of their path.

Reply:
I change the path router. That also fixes a problem when using hashed router.

```
if (!location.hash) {
	location.hash = '#/home';
}
```

is no longer needed